### PR TITLE
#66 create reusable apptoggle component

### LINKED ISSUE #66: Create reusable AppToggle component
# Task 🔨

We currently have two separate toggle components that were based on preliminary designs. Now that the designs are finalized, we should consolidate them into one `AppToggle` component. I think the current [AppToggleInternalText](https://github.com/ubclaunchpad/HappyHour/blob/a1799ae8e692c117975364600786ffd19d87b29f/src/common/AppToggleInternalText.vue) component is a good starting point for this.

In terms of API, something like the following could work for now:

```vue
<AppToggle v-model="isChecked" left-text="checked" right-text="not checked" />
```

Let's define the toggle to be "checked" if the **right text is active**. 

> Ideally the v-model should be based on some specified labels but in the interest of simplicity having it be checked-not checked is ok for now.

I believe the AppToggleInternalText component we have currently already implements most of this, but we should make sure that it's flexible enough that it can be used in multiple different screens.

# Resources ✏️ 

For reference, here's what it should look like:

CleanShot 2021-02-02 at 21 52 20@2x

### PR DESCRIPTION
**Pull request by baojill**
## Overview ⚙️

Modified AppInternalToggle component

## Change Summary

* AppInternalToggle now takes in isChecked prop that is "checked" when right text is active
* AppInternalToggle toggles between Single and Recurring event views

## Screenshots

Alt Text

## Related Issues ⚠️

* the toggle width is hardcoded right now and I'm not sure how I can use flex to un-hardcode that (apologies in advance for a lot of css)

Closes #66

### CODE REVIEW

File: src/common/AppToggle.vue
Comment (nandanmen) [inline src/common/AppToggle.vue:9]: I liked that you used an actual checkbox here!

File: src/common/AppToggle.vue
Comment (nandanmen) [inline src/common/AppToggle.vue]: I think this should be changed to `AppToggle` to match the file name (alternatively you can remove it altogether and let Vue infer the name automatically).

File: src/common/AppToggle.vue
Comment (nandanmen) [inline src/common/AppToggle.vue]: Is this method used anywhere? I don't see it anywhere — not sure if I'm missing something.

File: src/common/AppToggle.vue
Comment (nandanmen) [inline src/common/AppToggle.vue:13]: I changed these two to have the same classes since visually they look the same.

File: src/common/AppToggle.vue
Comment (nandanmen) [inline src/common/AppToggle.vue:45]: I added a `min-width` so the text doesn't look squished — the element can still grow larger with longer texts.

File: src/common/AppToggle.vue
Comment (nandanmen) [inline src/common/AppToggle.vue:49]: The key here is this `flex: 1` property.

When you assign multiple flex elements the same `flex` number, those elements will always take up the same proportion of space. Think of it like fractions — with two elements having `flex: 1`, each element will take up `1 (the flex number) / 2 (total elements)` space. When you have three elements, each element will take up `1/3` space and so on.

In this case, the elements will equally take up 50% of the container's space.

For more: https://developer.mozilla.org/en-US/docs/Web/CSS/flex

File: src/common/AppToggle.vue
Comment (nandanmen) [inline src/common/AppToggle.vue:78]: As a tip, you can make the `border-radius` a very large number to have the sides rounded but not stretched :)

File: src/common/AppToggle.vue
Comment (nandanmen) [inline src/common/AppToggle.vue:89]: Changed the dimensions to percentage-based so it'll change as the container changes.

**Review by nandanmen — APPROVED** [review]
@baojill Good start! I made the changes using flexbox, see my comments for specifics :)

I also changed the colors to match the theme as well as some minor tweaks to the animation. This is good to merge now once someone else approves it.

File: src/common/AppToggle.vue
Comment (jillxi) [reply src/common/AppToggle.vue:49]: thank you! I will definitely take a look at the document

**Review by tnwin — APPROVED** [review]

**Review by jamesmengo — APPROVED** [review]

**Pull request merged**
